### PR TITLE
Contribution refactor

### LINKED ISSUE #119: Contribution is unclear
it is called calculateContribution but returns the amount after deducting the contribution
in all places the resulting amount is still referred to as contribution amount
1. calculateContribution should return the actual contribution ie if contribution is 20% and amount is 100 it should return 20
2. in marketmaker it should subtract the contribution from the amount
3. the unit test for calculatecontribution of 20% is also unclear, it is just hard coded "8000".
add comment with explanation why is it 8000

### PR DESCRIPTION
**Pull request by ordd**
# Description

1. calculateContribution returns the actual contribution ie if contribution is 20% and amount is 100 it returns 20
2. marketmaker subtracts the contribution from the amount
3. the unit test for calculatecontribution of 20% updated.

About # (link your issue here)
#119 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request ( for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: stakingModel/contracts/GoodMarketMaker.sol [hunk 1]
```diff
@@ -262,7 +262,8 @@ contract GoodMarketMaker is BancorFormula, DSMath, SchemeGuard {
         require(_gdAmount >= _contributionGdAmount, "GD amount is lower than the contribution amount");
         ReserveToken storage rtoken = reserveTokens[address(_token)];
         require(rtoken.gdSupply > _gdAmount, "GD amount is higher than the total supply");
-        uint256 tokenReturn = sellReturn(_token, _contributionGdAmount);
+        uint256 deudcedAmount = _gdAmount.sub(_contributionGdAmount);
```
Comment (sirpy) [inline stakingModel/contracts/GoodMarketMaker.sol]:
```suggestion
        uint256 amountAfterContribution = _gdAmount.sub(_contributionGdAmount);
```

File: stakingModel/contracts/GoodMarketMaker.sol [hunk 2]
```diff
@@ -262,7 +262,8 @@ contract GoodMarketMaker is BancorFormula, DSMath, SchemeGuard {
...
+        uint256 tokenReturn = sellReturn(_token, deudcedAmount);
```
Comment (sirpy) [inline stakingModel/contracts/GoodMarketMaker.sol]:
```suggestion
        uint256 tokenReturn = sellReturn(_token, amountAfterContribution);
```

File: stakingModel/test/GoodReserveCDai.test.js [hunk 3]
```diff
@@ -445,11 +445,12 @@ it("should set marketmaker in the reserve by avatar", async () => {
     let nom = new BN(2e14).toString();
     let denom = new BN(1e15).toString();
     let actual = await contribution.calculateContribution(marketMaker.address, goodReserve.address, founder, cDAI.address, 1e4);
-    expect(actual.toString()).to.be.equal((1e4 - 1e4 * nom / denom).toString());
+    expect(actual.toString()).to.be.equal((1e4 * nom / denom).toString());
   });
 
   it("should be able to sell gd to cDAI with contribution of 20%", async () => {
     let amount = 1e4;
+    let expectedReturn = 800000; // deduced amount, ie. return minus contribution
```
Comment (sirpy) [inline stakingModel/test/GoodReserveCDai.test.js]:
```suggestion
    let expectedReturn = 800000; // return minus contribution = 1e4 GD = 1000000 * 80% = 800000
```

File: stakingModel/test/GoodReserveCDai.test.js [hunk 4]
```diff
@@ -445,11 +445,12 @@ it("should set marketmaker in the reserve by avatar", async () => {
     let nom = new BN(2e14).toString();
     let denom = new BN(1e15).toString();
     let actual = await contribution.calculateContribution(marketMaker.address, goodReserve.address, founder, cDAI.address, 1e4);
-    expect(actual.toString()).to.be.equal((1e4 - 1e4 * nom / denom).toString());
+    expect(actual.toString()).to.be.equal((1e4 * nom / denom).toString());
   });
 
   it("should be able to sell gd to cDAI with contribution of 20%", async () => {
     let amount = 1e4;
+    let expectedReturn = 800000; // return minus contribution = 1e4 GD = 1000000 * 80% = 800000
```
Comment (sirpy) [inline stakingModel/test/GoodReserveCDai.test.js]: I still dont understand this. Is the reserve ratio 1 here?
if we are selling 10k GD why do we expected 8K cDAI?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: stakingModel/test/GoodReserveCDai.test.js (hunk 4 above)
Comment (ordd) [reply stakingModel/test/GoodReserveCDai.test.js]: yes the rr is 100%
added more details
18399f793dd34475cf277effc433a76f204b0070

**Pull request merged**